File: java/src/main/java/codingdojo/Square.java
package codingdojo;

public class Square {
    public final char letter;
    public final int number;

    public Square(char letter, int number) {
        this.letter = letter;
        this.number = number;
    }

    public Square left() {
        return new Square((char)(letter+1), number);
    }

    public Square down() {
        return new Square(letter, number+1);
    }

    public String toModifier() {
        return new StringBuilder().append(letter).append(',').append(number).toString();
    }
}

File: java/src/main/java/codingdojo/Square.kt
package codingdojo

class Square(val letter: Char, val number: Int) {

    fun right(): Square {
        return Square(letter + 1, number)
    }

    fun down(): Square {
        return Square(letter, number + 1)
    }

    private fun getDirectNeighboor(direction: WordDirection): Square {
        return direction.next(this)
    }

    fun toCoordinates(): String {
        return "$letter,$number"
    }

    fun getNeighboor(direction: WordDirection, distance: Int): Square {
        if (distance == 0) return this
        else return getDirectNeighboor(direction).getNeighboor(direction, distance - 1)
    }
}